### PR TITLE
Updated

### PR DESCRIPTION
**Pull request by nabilwardeh**

### CODE REVIEW

File: web/src/components/views/Feed.js [hunk 1]
```diff
@@ -45,36 +46,69 @@ class Feed extends Component {
...
+        //         "Content-Type": "application/json"
+        //       },
+        //       credentials: "same-origin"
+        //     });
+        //     resp = await resp.json();
+        //     post.comments.push(...resp.comments);
+        //     return post;
+        //   })
+        // );
+        this.getComments(response);
```
Comment (kkarimi) [inline web/src/components/views/Feed.js]: You don't really need to pass the posts to the other function as you are storing the whole feed in in the `State` in the line just below this

File: web/src/components/views/Feed.js [hunk 2]
```diff
@@ -45,36 +46,69 @@ class Feed extends Component {
...
           feed: response,
           loadLastComments: []
         });
-        this.forceUpdate();
       });
   };
+
+  getComments = (posts) => {
+    console.log('FETCHING COMMENTS');
+    let addPostComments = posts.map(post => {
```
Comment (kkarimi) [inline web/src/components/views/Feed.js]: Not sure I understand the reason for defining addPostComments? the usual pattern is to 1) get posts from the state, then 2) process it (add comments) and 3) at the end of the function update the `State` with the new posts which includes comments.

File: web/src/components/views/Feed.js [hunk 3]
```diff
@@ -45,36 +46,69 @@ class Feed extends Component {
...
+      fetch(`/api/post/${post.postID}`, {
+        method: "GET",
+        headers: {
+          Accept: "application/json",
+          "Content-Type": "application/json"
+        },
+        credentials: "same-origin"
```
Comment (kkarimi) [inline web/src/components/views/Feed.js]: This should carry the session to the next request, so can't see the reason for `401`, might want to put a debugger to see where/why the session info goes missing from the request

File: web/src/components/views/Feed.js (hunk 2 above)
Comment (nabilwardeh) [reply web/src/components/views/Feed.js]: This function could have been added to the component DidMount function but that will make it a very long one.

File: web/src/components/views/Feed.js (hunk 3 above)
Comment (nabilwardeh) [reply web/src/components/views/Feed.js]: As I said it is happening for all request. Again, I cloned the original repo from CYF to a new foldeR on my machine and I am facing the same issue. It is not about my code. It seems something happened and it is resulting in this issue. The only change was installing yarn which I uninstalled but the problem persists.

File: web/src/components/views/Feed.js (hunk 2 above)
Comment (kkarimi) [reply web/src/components/views/Feed.js]: I was referring to the `addPostComments` variable

**Pull request merged**